common/persistence/sql/sqlHistoryStore.go

```diff
@@ -434,9 +440,20 @@ func (m *sqlHistoryStore) DeleteHistoryBranch(
 				// No any branch is using this range, we can delete all of it
 				nodeFilter.MinNodeID = br.BeginNodeID
 			}
-			_, err := tx.DeleteFromHistoryNode(ctx, nodeFilter)
-			if err != nil {
-				return err
+			for {
+				result, err := tx.DeleteFromHistoryNode(ctx, nodeFilter)
+				if err != nil {
+					return err
+				}
+				rowsAffected, err := result.RowsAffected()
+				if err != nil {
+					return err
+				}
+				if rowsAffected < _defaultHistoryNodeDeleteBatch ||
+					rowsAffected == persistence.UnknownNumRowsAffected ||
+					rowsAffected > _defaultHistoryNodeDeleteBatch {
```

Comment:
nit, align the condition lines with the same indentation.

Reply:
How's `rowsAffected > _defaultHistoryNodeDeleteBatch` possible?

Reply:
We allow pagesize not to be honored. In case, there is some database doesn't support Delete Order By Limit.